test/uk/gov/hmrc/trusts/controllers/RegisterTrustControllerSpec.scala

```diff
@@ -16,106 +16,171 @@
 
 package uk.gov.hmrc.trusts.controllers
 
+import org.mockito.Matchers
 import org.mockito.Matchers.any
-import org.mockito.Mockito.when
+import org.mockito.Mockito.{when, _}
 import org.scalatestplus.play.guice.GuiceOneServerPerSuite
 import play.api.libs.json.Json
 import play.api.test.Helpers.{status, _}
-import uk.gov.hmrc.auth.core.{BearerTokenExpired, MissingBearerToken}
+import uk.gov.hmrc.auth.core.{BearerTokenExpired, MissingBearerToken, _}
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.trusts.config.AppConfig
 import uk.gov.hmrc.trusts.connectors.{BaseSpec, FakeAuthConnector}
 import uk.gov.hmrc.trusts.exceptions._
 import uk.gov.hmrc.trusts.models._
-import uk.gov.hmrc.trusts.services.{AuthService, DesService, ValidationService}
+import uk.gov.hmrc.trusts.services.{AuthService, DesService, RosmPatternService, ValidationService}
 
 import scala.concurrent.Future
 
 
 class RegisterTrustControllerSpec extends BaseSpec with GuiceOneServerPerSuite {
 
   val mockDesService = mock[DesService]
+  val rosmPatternService = mock[RosmPatternService]
+  val authConnector = mock[AuthConnector]
 
   lazy val appConfig: AppConfig = app.injector.instanceOf[AppConfig]
   lazy val validationService: ValidationService = new ValidationService()
 
+  before {
+    reset(rosmPatternService)
+  }
+
   ".registration" should {
 
     "return 200 with TRN" when {
-      "the register endpoint is called with a valid json payload " in {
-        val mockAuthService = new AuthService(FakeAuthConnector())
-        val SUT = new RegisterTrustController(mockDesService, appConfig, validationService, mockAuthService)
+      "individual user called the register endpoint with a valid json payload " in {
+        when(mockDesService.registerTrust(any[Registration])(any[HeaderCarrier]))
+          .thenReturn(Future.successful(RegistrationTrustResponse("XTRN123456")))
```

Comment:
Repeated us of "XTRN123456" ... can we make this a constant val ?

Reply:
Done @jassalrichy